docs/developer/mesh.md

```diff
@@ -0,0 +1,270 @@
+# Mesh
+
+Here we describe the two mesh formarts that are used in RDycore. The two mesh
+formats that we use are:
+
+1. Exodus II
+2. DMPlex HDF5 v3.0.0
+
+RDycore has been tested for meshes that include triangular, quadrilateral, or
+both triangular and quadrilateral cells. The vertices of cells have coordinates
+in 3D.
+
+## Exodus II Mesh Format
+
+The Exodus II mesh format (`.exo`) uses 1-based indices.
+
+### Triangular elements
+
+```text
+     Triangular
+        cell
+         v3
+         / \
+        /   \
+      e5     e4
+     /   c1   \
+    /          \
+  v1 --- e3 --- v2
+```
+
+- The `elem_type` of the triangular cell, `c1`, is `TRI3`
+- It comprises of three vertices (i.e. `v1`, `v2`, `v3`)
+- It has five edges sets. The first two edge sets form a plane and the remaining
+  three edge sets are lines as follows:
+  - `e1`: Comprises of the plane formed by `v1`, `v2`, `v3` (not shown above)
+  - `e2`: Comprises of the plane formed by `v1`, `v3`, `v2` (not shown above)
+  - `e3`: Line from `v1` to `v2`
+  - `e4`: Line from `v2` to `v3`
+  - `e5`: Line from `v3` to `v1`
```

Comment:
If edges are directed (i.e. if edge `e3` starts at `v1` and ends at `v2`), maybe we should call them "directed line segments" instead of "lines", which aren't directed and technical extend infinitely in either direction.

Reply:
Now uses the term "_a directed line_"